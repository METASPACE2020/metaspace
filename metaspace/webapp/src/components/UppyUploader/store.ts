// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from 'vue'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State | any>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach((listener) => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
