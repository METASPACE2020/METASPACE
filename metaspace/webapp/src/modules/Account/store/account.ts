import { Module } from 'vuex';
import { DialogType } from '../dialogs';
import router from '../../../router';

export interface AccountState {
  dialog: DialogType | null;
  showDialogAsPage: boolean;
}

const account = {
  namespaced: true,
  state: {
    dialog: null,
    showDialogAsPage: false
  },
  mutations: {
    showDialog(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = false;
    },
    showDialogAsPage(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = true;
    },
    hideDialog(state: AccountState, dialog: DialogType) {
      if (state.dialog === dialog) {
        state.dialog = null;
        if (state.showDialogAsPage) {
          router.push('/');
        }
      }
    }
  }
};

export default account as Module<AccountState, any>;
