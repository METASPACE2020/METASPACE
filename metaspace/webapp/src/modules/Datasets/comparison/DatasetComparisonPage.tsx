import { Collapse, CollapseItem } from '../../../lib/element-ui'
import {
  computed,
  defineComponent,
  onMounted, reactive,
  ref, watchEffect,
} from '@vue/composition-api'
import { useQuery, useMutation } from '@vue/apollo-composable'
import { comparisonAnnotationListQuery } from '../../../api/annotation'
import safeJsonParse from '../../../lib/safeJsonParse'
import RelatedMolecules from '../../Annotations/annotation-widgets/RelatedMolecules.vue'
import ImageSaver from '../../ImageViewer/ImageSaver.vue'
import { DatasetComparisonAnnotationTable } from './DatasetComparisonAnnotationTable'
import { DatasetComparisonGrid } from './DatasetComparisonGrid'
import gql from 'graphql-tag'
import FilterPanel from '../../Filters/FilterPanel.vue'
import { isEqual } from 'lodash'
import config from '../../../lib/config'
import { cloneDeep } from 'lodash-es'

interface DatasetComparisonPageProps {
  className: string
  defaultImagePosition: any
}

interface DatasetComparisonPageState {
  selectedAnnotation: number
  gridState: any
  annotations: any
  grid: any
  nCols: number
  nRows: number
  annotationData: any
  refsLoaded: boolean
  showViewer: boolean
  annotationLoading: boolean
  filter: any
  isLoading: any
}

export default defineComponent<DatasetComparisonPageProps>({
  name: 'DatasetComparisonPage',
  props: {
    className: {
      type: String,
      default: 'dataset-comparison',
    },
    defaultImagePosition: {
      type: Object,
      default: () => ({
        zoom: 1,
        xOffset: 0,
        yOffset: 0,
      }),
    },
  },

  // @ts-ignore
  setup(props, { refs, root }) {
    const fetchImageViewerSnapshot = gql`query fetchImageViewerSnapshot($id: String!, $datasetId: String!) {
      imageViewerSnapshot(id: $id, datasetId: $datasetId) {
        snapshot
      }
    }`
    const { $route, $store } = root
    const gridNode = ref(null)
    const state = reactive<DatasetComparisonPageState>({
      selectedAnnotation: -1,
      gridState: {},
      annotations: [],
      grid: undefined,
      nCols: 0,
      nRows: 0,
      annotationData: {},
      refsLoaded: false,
      showViewer: false,
      annotationLoading: true,
      filter: $store.getters.filter,
      isLoading: false,
    })
    const { snapshot_id: snapshotId, dataset_id: sourceDsId } = $route.params
    const {
      result: settingsResult,
      loading: settingsLoading,
    } = useQuery<any>(fetchImageViewerSnapshot, {
      id: snapshotId,
      datasetId: sourceDsId,
    })

    const gridSettings = computed(() => settingsResult.value != null
      ? settingsResult.value.imageViewerSnapshot : null)

    const queryVariables = () => {
      const filter = $store.getters.gqlAnnotationFilter
      const dFilter = $store.getters.gqlDatasetFilter
      const colocalizationCoeffFilter = $store.getters.gqlColocalizationFilter
      const query = $store.getters.ftsQuery

      return {
        filter,
        dFilter,
        query,
        colocalizationCoeffFilter,
        countIsomerCompounds: config.features.isomers,
      }
    }

    const queryOptions = reactive({ enabled: false })
    const queryVars = computed(() => ({
      ...queryVariables(),
      dFilter: { ...queryVariables().dFilter, ids: Object.values(state.grid || {}).join('|') },
    }))
    const annotationsQuery = useQuery<any>(comparisonAnnotationListQuery, queryVars, queryOptions)
    const loadAnnotations = () => { queryOptions.enabled = true }
    state.annotations = computed(() => {
      if (annotationsQuery.result.value) {
        return annotationsQuery.result.value.allAggregatedAnnotations
      }
      return null
    })

    const requestAnnotations = async() => {
      state.isLoading = true
      loadAnnotations()
      state.annotationLoading = false
      state.isLoading = false
    }

    onMounted(() => {
      state.refsLoaded = true
    })

    watchEffect(async() => {
      if (!state.grid && gridSettings.value) {
        const auxSettings = safeJsonParse(gridSettings.value.snapshot)
        state.nCols = auxSettings.nCols
        state.nRows = auxSettings.nRows
        state.grid = auxSettings.grid
        await requestAnnotations()
      }
    })

    const handleRowChange = (idx: number) => {
      if (idx !== -1) {
        state.isLoading = true
        state.selectedAnnotation = idx
        setTimeout(() => {
          state.isLoading = false
        }, 500)
      }
    }

    const renderImageGallery = (nCols: number, nRows: number) => {
      return (
        <CollapseItem
          id="annot-img-collapse"
          name="images"
          title="Image viewer"
          class="ds-collapse el-collapse-item--no-padding relative">
          <ImageSaver
            class="absolute top-0 right-0 mt-2 mr-2 dom-to-image-hidden"
            domNode={gridNode.value}
          />
          <div class='dataset-comparison-grid' ref={gridNode}>
            <DatasetComparisonGrid
              nCols={nCols}
              nRows={nRows}
              settings={gridSettings}
              annotations={state.annotations || []}
              selectedAnnotation={state.selectedAnnotation}
              isLoading={state.isLoading}
            />
          </div>
        </CollapseItem>)
    }

    const renderCompounds = () => {
      // @ts-ignore TS2604
      const relatedMolecules = (annotation: any) => <RelatedMolecules
        query="isomers"
        annotation={annotation}
        databaseId={$store.getters.filter.database || 1}
        hideFdr
      />

      return (<CollapseItem
        id="annot-img-collapse"
        name="compounds"
        title="Molecules"
        class="ds-collapse el-collapse-item--no-padding relative">
        {
          state.annotations
          && state.annotations[state.selectedAnnotation]
          && state.annotations[state.selectedAnnotation].annotations.map((ds: any) => {
            return relatedMolecules(ds)
          })
        }
      </CollapseItem>)
    }

    return () => {
      const nCols = state.nCols
      const nRows = state.nRows

      return (
        <div class='dataset-comparison-page w-full flex flex-wrap flex-row'>
          <FilterPanel class='w-full' level='annotation' hiddenFilters={['datasetIds']}/>
          <div class='dataset-comparison-wrapper w-full md:w-5/12'>
            <DatasetComparisonAnnotationTable
              isLoading={state.annotationLoading}
              annotations={(cloneDeep(state.annotations || [])).map((ion: any) => ion.annotations[0])}
              onRowChange={handleRowChange}/>
          </div>
          <div class='dataset-comparison-wrapper  w-full  md:w-7/12'>
            <Collapse value={'images'} id="annot-content"
              class="border-0">
              {renderImageGallery(nCols, nRows)}
              {renderCompounds()}
            </Collapse>
          </div>
        </div>
      )
    }
  },
})
