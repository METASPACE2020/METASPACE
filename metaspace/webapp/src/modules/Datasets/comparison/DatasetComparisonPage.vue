<script src="./DatasetComparisonPage.tsx" />
<style src="./DatasetComparisonPage.scss" lang="scss" />
