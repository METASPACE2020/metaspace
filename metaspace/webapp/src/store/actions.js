import { fetchOptionListsQuery } from '../api/metadata'
import { decodeParams } from '../modules/Filters'
import { computed } from 'vue'
import store from '../store/index'

export default {
  async initFilterLists(context, apolloClient) {
    if (context.state.filterListsLoading || context.state.filterLists != null) return
    await context.commit('setFilterListsLoading')

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    })

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {
      ...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds),
    })

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)

    context.commit('updateFilter', filter)
  },
}
