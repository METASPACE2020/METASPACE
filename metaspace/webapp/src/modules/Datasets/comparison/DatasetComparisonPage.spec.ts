import { mount } from '@vue/test-utils'
import router from '../../../router'
import store from '../../../store/index'
import Vue from 'vue'
import Vuex from 'vuex'
import { sync } from 'vuex-router-sync'
import DatasetComparisonPage from './DatasetComparisonPage'
import { initMockGraphqlClient, apolloProvider } from '../../../../tests/utils/mockGraphqlClient'

describe('DatasetComparisonPage', () => {
  const snapshotData = jest.fn((src: any, args: any, ctx: any) => ({
    snapshot: '{"nCols":2,"nRows":1,"grid":{"0-0":"2021-04-14_07h23m35s",'
      + '"0-1":"2021-04-06_08h35m04s"}}',
  }))

  const annotationsData : any = {
    allAggregatedAnnotations: [
      {
        id: '2021-03-2021-04-14_07h23m35s',
        sumFormula: 'C10H11NO',
        ion: 'C10H11NO+Na+',
        ionFormula: 'C10H11NONa',
        database: 'HMDB',
        msmScore: 0.736404,
        rhoSpatial: 0.848534,
        rhoSpectral: 0.982476,
        rhoChaos: 0.883333,
        fdrLevel: 0.05,
        mz: 184.07324581966384,
        colocalizationCoeff: null,
        offSample: null,
        offSampleProb: null,
        isomers: [],
        isobars: [],
        countPossibleCompounds: 5,
        possibleCompounds: [],
      },
      {
        id: '2021-04-06_08h35m04s',
        sumFormula: 'C23H45NO4',
        adduct: '+H',
        ion: 'C23H45NO4+H+',
        ionFormula: 'C23H46NO4',
        database: 'HMDB',
        msmScore: 0.509524,
        rhoSpatial: 0.577076,
        rhoSpectral: 0.990214,
        rhoChaos: 0.891667,
        fdrLevel: 0.05,
        mz: 400.34209613966374,
        colocalizationCoeff: null,
        offSample: null,
        offSampleProb: null,
        isotopeImages: [
          {
            mz: 400.34209613966374,
            url: 'http://storage:9000/sm-ion-images-dev/iso/2021-03-31_11h02m28s/fd53ff218cfb420f988b9fc0723b802a',
            minIntensity: 8137228,
            maxIntensity: 11610989,
            totalIntensity: 76451200,
          },
        ],
        isomers: [],
        isobars: [],
        countPossibleCompounds: 1,
        possibleCompounds: [],
      },
    ],
  }

  const testHarness = Vue.extend({
    components: {
      DatasetComparisonPage,
    },
    render(h) {
      return h(DatasetComparisonPage, { props: this.$attrs })
    },
  })

  const graphqlWithData = () => {
    initMockGraphqlClient({
      Query: () => ({
        imageViewerSnapshot: snapshotData,
      })
    })
  }

  beforeAll(() => {
    Vue.use(Vuex)
    sync(store, router)
    router.replace({
      name: 'datasets-comparison',
      params: {
        snapshot_id: 'xxxx',
        dataset_id: 'xxxx',
      },
    })
  })

  it('it should match snapshot', async() => {
    graphqlWithData()
    const wrapper = mount(testHarness, {
      store,
      router,
    })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('it should call snapshot settings query', async() => {
    graphqlWithData()
    mount(testHarness, { store, router })
    await Vue.nextTick()

    expect(snapshotData).toHaveBeenCalledTimes(1)
  })
})
