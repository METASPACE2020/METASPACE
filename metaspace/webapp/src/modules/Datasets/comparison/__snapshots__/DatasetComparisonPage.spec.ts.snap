// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetComparisonPage it should match snapshot 1`] = `
<div class="dataset-comparison-page w-full flex flex-wrap flex-row">
  <div class="filter-panel w-full">
    <mock-el-select placeholder="Add filter" class="filter-select">
      <mock-el-option value="group" label="Select group"></mock-el-option>
      <mock-el-option value="project" label="Select project"></mock-el-option>
      <mock-el-option value="submitter" label="Select submitter"></mock-el-option>
      <mock-el-option value="datasetIds" label="Select dataset"></mock-el-option>
      <mock-el-option value="compoundName" label="Search molecule"></mock-el-option>
      <mock-el-option value="mz" label="Search by m/z"></mock-el-option>
      <mock-el-option value="offSample" label="Show/hide off-sample annotations"></mock-el-option>
      <mock-el-option value="polarity" label="Select polarity"></mock-el-option>
      <mock-el-option value="adduct" label="Select adduct"></mock-el-option>
      <mock-el-option value="organism" label="Select organism"></mock-el-option>
      <mock-el-option value="organismPart" label="Select organism part"></mock-el-option>
      <mock-el-option value="condition" label="Select organism condition"></mock-el-option>
      <mock-el-option value="growthConditions" label="Select sample growth conditions"></mock-el-option>
      <mock-el-option value="analyzerType" label="Select analyzer"></mock-el-option>
      <mock-el-option value="ionisationSource" label="Select ionisation source"></mock-el-option>
      <mock-el-option value="maldiMatrix" label="Select MALDI matrix"></mock-el-option>
      <mock-el-option value="minMSM" label="Set minimum MSM score"></mock-el-option>
    </mock-el-select>
    <div class="tf-outer w-auto el-input el-input-group el-input-group--prepend" data-test-key="simpleQuery" filter-key="simpleQuery" name="Simple query" options="">
      <div class="el-input-group__prepend"><i class="el-icon-search -mx-1"></i></div><input type="text" autocomplete="off" placeholder="Enter keywords" class="el-input__inner">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="tf-outer border-gray-300 border border-solid text-sm pr-3" data-test-key="database" filter-key="database" name="Database" options="">
      <div class="tf-name bg-gray-100 text-gray-700 tracking-tight px-3 border-0 border-r border-solid border-gray-300">
        Database:
        <!---->
      </div>
      <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container pl-3">
        <div slot-key="default">
          <div>
            <mock-el-select placeholder="Start typing name" filterable="" filter-method="function () { [native code] }" no-data-text="No matches" no-match-text="No matches" reserve-keyword="" class="w-full"></mock-el-select>
            <p class="leading-5 text-sm m-0 mt-3 text-gray-700"><i class="text-gray-600 mr-1 el-icon-info"></i>
              Search to see archived versions
            </p>
          </div>
        </div>
        <div slot-key="reference">
          <div class="tf-value"><i class="el-icon-loading"></i>
            <!---->
          </div>
        </div>
      </mock-el-popover>
      <!---->
    </div>
    <div class="tf-outer border-gray-300 border border-solid text-sm pr-3" data-test-key="fdrLevel" filter-key="fdrLevel">
      <div class="tf-name bg-gray-100 text-gray-700 tracking-tight px-3 border-0 border-r border-solid border-gray-300">
        FDR:
        <!---->
      </div>
      <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container pl-3">
        <div slot-key="default">
          <mock-el-select value="0.1">
            <mock-el-option label="5%" value="0.05"></mock-el-option>
            <mock-el-option label="10%" value="0.1"></mock-el-option>
            <mock-el-option label="20%" value="0.2"></mock-el-option>
            <mock-el-option label="50%" value="0.5"></mock-el-option>
          </mock-el-select>
        </div>
        <div slot-key="reference">
          <div class="tf-value"><span class="tf-value-span"><span>10%</span></span>
            <!---->
          </div>
        </div>
      </mock-el-popover>
      <!---->
    </div>
    <div class="tf-outer border-gray-300 border border-solid text-sm pr-3" data-test-key="metadataType" filter-key="metadataType">
      <div class="tf-name bg-gray-100 text-gray-700 tracking-tight px-3 border-0 border-r border-solid border-gray-300">
        Data type:
        <!---->
      </div>
      <mock-el-popover trigger="click" placement="bottom" width="300" class="tf-value-container pl-3">
        <div slot-key="default">
          <mock-el-select filterable="true" value="Imaging MS">
            <mock-el-option label="Imaging MS" value="Imaging MS"></mock-el-option>
            <mock-el-option label="LC-MS" value="LC-MS"></mock-el-option>
          </mock-el-select>
        </div>
        <div slot-key="reference">
          <div class="tf-value"><span class="tf-value-span"><span>Imaging MS</span></span>
            <!---->
          </div>
        </div>
      </mock-el-popover>
      <!---->
    </div>
  </div>
  <div class="dataset-comparison-wrapper w-full md:w-5/12">
    <div class="ds-comparison-annotation-table-loading-wrapper"><i class="el-icon-loading"></i></div>
  </div>
  <div class="dataset-comparison-wrapper  w-full  md:w-7/12">
    <mock-el-collapse value="images" id="annot-content" class="border-0">
      <mock-el-collapse-item id="annot-img-collapse" name="images" title="Image viewer" class="ds-collapse el-collapse-item--no-padding relative"><button title="Export image" class="button-reset rounded-full bg-gray-100 shadow-xs h-8 w-8 flex items-center justify-center cursor-not-allowed absolute top-0 right-0 mt-2 mr-2 dom-to-image-hidden"><i class="el-icon-download text-xl"></i></button>
        <div class="dataset-comparison-grid">
          <div class="dataset-comparison-grid">
            <div class="dataset-comparison-grid-row">
              <div class="dataset-comparison-grid-col overflow-hidden"><span slot="title" class="w-full dataset-comparison-grid-item-header dom-to-image-hidden"><span>
    Image viewer
  </span>
                <div class="flex items-center pr-4 cursor-default">
                  <mock-el-popover placement="bottom" trigger="click">
                    <div slot-key="default"><span id="ion-image-settings"><form class="el-form el-form--label-top"><div class="el-form-item" data-feature-anchor="ion-image-color-scale"><label class="el-form-item__label">Intensity scaling</label><div class="el-form-item__content"><mock-el-select value="linear" style="width: 300px;"><mock-el-option value="linear" label="Linear"></mock-el-option> <mock-el-option value="linear-full" label="Linear (without hot-spot removal)"></mock-el-option> <mock-el-option value="log" label="Logarithmic"></mock-el-option> <mock-el-option value="log-full" label="Logarithmic (without outlier clipping)"></mock-el-option> <mock-el-option value="hist" label="Equalized histogram"></mock-el-option></mock-el-select><transition-stub name="el-zoom-in-top"><!----></transition-stub></div></div> <div class="el-form-item"><label class="el-form-item__label">Colormap</label><div class="el-form-item__content"><mock-el-select value="Viridis" style="width: 150px;"><ul class="el-select-group__wrap"><li class="el-select-group__title"></li><li><ul class="el-select-group"><mock-el-option value="Viridis" label="Viridis" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="Cividis" label="Cividis" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="Hot" label="Hot" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="YlGnBu" label="YlGnBu" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="Portland" label="Portland" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="Greys" label="Greys" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option></ul></li></ul> <ul class="el-select-group__wrap"><li class="el-select-group__title">Inverted</li><li><ul class="el-select-group"><mock-el-option value="-Viridis" label="Viridis" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-Cividis" label="Cividis" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-Hot" label="Hot" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-YlGnBu" label="YlGnBu" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-Portland" label="Portland" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-Greys" label="Greys" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option></ul></li></ul></mock-el-select><transition-stub name="el-zoom-in-top"><!----></transition-stub></div></div> <div class="el-form-item"><label class="el-form-item__label">Scale bar color</label><div class="el-form-item__content"><mock-el-select value="#000000" style="width: 150px;"><mock-el-option value="#000000" label="Black"><div style="position: relative; background: transparent; height: 50px;"><div class="scale-bar"><div class="scale-bar-x" style="color: rgb(0, 0, 0); border-color: #000000; width: 62.5px;"><div class="scale-bar-x-text">
      500 Âµm
    </div></div> <!----></div></div></mock-el-option><mock-el-option value="#999999" label="Gray"><div style="position: relative; background: transparent; height: 50px;"><div class="scale-bar"><div class="scale-bar-x" style="color: rgb(153, 153, 153); border-color: #999999; width: 62.5px;"><div class="scale-bar-x-text">
      500 Âµm
    </div></div> <!----></div></div></mock-el-option><mock-el-option value="#FFFFFF" label="White"><div style="position: relative; background: rgb(204, 204, 204); height: 50px;"><div class="scale-bar"><div class="scale-bar-x" style="color: rgb(255, 255, 255); border-color: #ffffff; width: 62.5px;"><div class="scale-bar-x-text">
      500 Âµm
    </div></div> <!----></div></div></mock-el-option> <mock-el-option value="hidden" label="Hidden"></mock-el-option></mock-el-select><transition-stub name="el-zoom-in-top"><!----></transition-stub></div></div></form></span> </div>
                    <div slot-key="reference"><button class="button-reset av-icon-button"><i data-feature-anchor="ion-image-settings" class="el-icon-setting" style="font-size: 20px; vertical-align: middle;"></i></button></div>
                  </mock-el-popover> <button title="Reset image zoom and offsets" class="button-reset av-icon-button"><svg data-icon="aspect-ratio" class="w-6 h-6 fill-current text-gray-900"></svg></button>
                  <!---->
                </div>
                <transition-stub mode="out-in" enterclass="opacity-0" leavetoclass="opacity-0" enteractiveclass="duration-150 transition-opacity ease-in-out" leaveactiveclass="duration-150 transition-opacity ease-in-out"></transition-stub>
                </span>
                <div class="ds-wrapper relative"><i class="el-icon-loading"></i>
                  <div class="ds-viewer-controls-wrapper  v-rhythm-3 sm-side-bar">
                    <transition-stub mode="out-in" enterclass="opacity-0" leavetoclass="opacity-0" enteractiveclass="duration-150 transition-opacity ease-in-out" leaveactiveclass="duration-150 transition-opacity ease-in-out" class="absolute bottom-0 right-0 mt-3 ml-3 dom-to-image-hidden"></transition-stub>
                    <transition-stub mode="out-in" enterclass="opacity-0" leavetoclass="opacity-0" enteractiveclass="duration-150 transition-opacity ease-in-out" leaveactiveclass="duration-150 transition-opacity ease-in-out" class="absolute top-0 right-0 mt-3 ml-3 dom-to-image-hidden"></transition-stub>
                  </div><button title="Export image" class="button-reset rounded-full bg-gray-100 shadow-xs h-8 w-8 flex items-center justify-center cursor-not-allowed absolute top-0 left-0 mt-3 ml-3 dom-to-image-hidden"><i class="el-icon-download text-xl"></i></button>
                </div>
              </div>
              <div class="dataset-comparison-grid-col overflow-hidden"><span slot="title" class="w-full dataset-comparison-grid-item-header dom-to-image-hidden"><span>
    Image viewer
  </span>
                <div class="flex items-center pr-4 cursor-default">
                  <mock-el-popover placement="bottom" trigger="click">
                    <div slot-key="default"><span id="ion-image-settings"><form class="el-form el-form--label-top"><div class="el-form-item" data-feature-anchor="ion-image-color-scale"><label class="el-form-item__label">Intensity scaling</label><div class="el-form-item__content"><mock-el-select value="linear" style="width: 300px;"><mock-el-option value="linear" label="Linear"></mock-el-option> <mock-el-option value="linear-full" label="Linear (without hot-spot removal)"></mock-el-option> <mock-el-option value="log" label="Logarithmic"></mock-el-option> <mock-el-option value="log-full" label="Logarithmic (without outlier clipping)"></mock-el-option> <mock-el-option value="hist" label="Equalized histogram"></mock-el-option></mock-el-select><transition-stub name="el-zoom-in-top"><!----></transition-stub></div></div> <div class="el-form-item"><label class="el-form-item__label">Colormap</label><div class="el-form-item__content"><mock-el-select value="Viridis" style="width: 150px;"><ul class="el-select-group__wrap"><li class="el-select-group__title"></li><li><ul class="el-select-group"><mock-el-option value="Viridis" label="Viridis" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="Cividis" label="Cividis" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="Hot" label="Hot" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="YlGnBu" label="YlGnBu" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="Portland" label="Portland" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="Greys" label="Greys" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option></ul></li></ul> <ul class="el-select-group__wrap"><li class="el-select-group__title">Inverted</li><li><ul class="el-select-group"><mock-el-option value="-Viridis" label="Viridis" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-Cividis" label="Cividis" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-Hot" label="Hot" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-YlGnBu" label="YlGnBu" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-Portland" label="Portland" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option><mock-el-option value="-Greys" label="Greys" class="flex items-center pr-5 py-2"><div class="h-full w-full"></div></mock-el-option></ul></li></ul></mock-el-select><transition-stub name="el-zoom-in-top"><!----></transition-stub></div></div> <div class="el-form-item"><label class="el-form-item__label">Scale bar color</label><div class="el-form-item__content"><mock-el-select value="#000000" style="width: 150px;"><mock-el-option value="#000000" label="Black"><div style="position: relative; background: transparent; height: 50px;"><div class="scale-bar"><div class="scale-bar-x" style="color: rgb(0, 0, 0); border-color: #000000; width: 62.5px;"><div class="scale-bar-x-text">
      500 Âµm
    </div></div> <!----></div></div></mock-el-option><mock-el-option value="#999999" label="Gray"><div style="position: relative; background: transparent; height: 50px;"><div class="scale-bar"><div class="scale-bar-x" style="color: rgb(153, 153, 153); border-color: #999999; width: 62.5px;"><div class="scale-bar-x-text">
      500 Âµm
    </div></div> <!----></div></div></mock-el-option><mock-el-option value="#FFFFFF" label="White"><div style="position: relative; background: rgb(204, 204, 204); height: 50px;"><div class="scale-bar"><div class="scale-bar-x" style="color: rgb(255, 255, 255); border-color: #ffffff; width: 62.5px;"><div class="scale-bar-x-text">
      500 Âµm
    </div></div> <!----></div></div></mock-el-option> <mock-el-option value="hidden" label="Hidden"></mock-el-option></mock-el-select><transition-stub name="el-zoom-in-top"><!----></transition-stub></div></div></form></span> </div>
                    <div slot-key="reference"><button class="button-reset av-icon-button"><i data-feature-anchor="ion-image-settings" class="el-icon-setting" style="font-size: 20px; vertical-align: middle;"></i></button></div>
                  </mock-el-popover> <button title="Reset image zoom and offsets" class="button-reset av-icon-button"><svg data-icon="aspect-ratio" class="w-6 h-6 fill-current text-gray-900"></svg></button>
                  <!---->
                </div>
                <transition-stub mode="out-in" enterclass="opacity-0" leavetoclass="opacity-0" enteractiveclass="duration-150 transition-opacity ease-in-out" leaveactiveclass="duration-150 transition-opacity ease-in-out"></transition-stub>
                </span>
                <div class="ds-wrapper relative"><i class="el-icon-loading"></i>
                  <div class="ds-viewer-controls-wrapper  v-rhythm-3 sm-side-bar">
                    <transition-stub mode="out-in" enterclass="opacity-0" leavetoclass="opacity-0" enteractiveclass="duration-150 transition-opacity ease-in-out" leaveactiveclass="duration-150 transition-opacity ease-in-out" class="absolute bottom-0 right-0 mt-3 ml-3 dom-to-image-hidden"></transition-stub>
                    <transition-stub mode="out-in" enterclass="opacity-0" leavetoclass="opacity-0" enteractiveclass="duration-150 transition-opacity ease-in-out" leaveactiveclass="duration-150 transition-opacity ease-in-out" class="absolute top-0 right-0 mt-3 ml-3 dom-to-image-hidden"></transition-stub>
                  </div><button title="Export image" class="button-reset rounded-full bg-gray-100 shadow-xs h-8 w-8 flex items-center justify-center cursor-not-allowed absolute top-0 left-0 mt-3 ml-3 dom-to-image-hidden"><i class="el-icon-download text-xl"></i></button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </mock-el-collapse-item>
      <mock-el-collapse-item id="annot-img-collapse" name="compounds" title="Molecules" class="ds-collapse el-collapse-item--no-padding relative"></mock-el-collapse-item>
    </mock-el-collapse>
  </div>
</div>
`;
